### DISCUSSION TITLE
SSCHA tutorial 2023: Tutorial 7: electron phonon coupling constant calculation

### DISCUSSION THREAD

**Discussion by ArchSony** [SSCHAcode/python-sscha#347] [Q&A] [answered]
While following the tutorial, in the SSCHA CALCULATION STEP to obtain the "sscha_T0.0_dyn" matrix I am following the python code (sscha_Au.py from tutorial 1). These are the errors i am getting.  
Immaginary frequencies found! Redoing the step.

Error, exceeded the maximum number of step with an imaginary frequency (50).
       it is possible that something wrong is happening with the original dynamical matrix.
       This error may be caused by a wrong imposition of acoustic sum rule or
       the presence of 0 frequency modes.

       Final dynamical matrix saved into error_dyn.

       If you believe this is not a problem, you can try to increase max_imag_freq_counter
       from the 'SSCHA_Minimizer' object or reduce the 'min_step_dyn'

Traceback (most recent call last):
  File "/home/amritendu/archana/SMO/pdh_qe/222/sscha_1/sscha_PdH1.py", line 76, in <module>
    relax.relax(get_stress = True)
  File "/home/amritendu/.local/lib/python3.10/site-packages/sscha/Relax.py", line 388, in relax
    self.minim.run(custom_function_pre = self.__cfpre__,
  File "/home/amritendu/.local/lib/python3.10/site-packages/sscha/SchaMinimizer.py", line 1298, in run
    timer.execute_timed_function(self.minimization_step, custom_function_gradient)
  File "/home/amritendu/.local/lib/python3.10/site-packages/cellconstructor/Timer.py", line 112, in execute_timed_function
    ret = function(*args, timer=new_timer,**kwargs)
  File "/home/amritendu/.local/lib/python3.10/site-packages/sscha/SchaMinimizer.py", line 702, in minimization_step
    raise ValueError(ERROR_MSG)
ValueError: 
Error, exceeded the maximum number of step with an imaginary frequency (50).
       it is possible that something wrong is happening with the original dynamical matrix.
       This error may be caused by a wrong imposition of acoustic sum rule or
       the presence of 0 frequency modes.

       Final dynamical matrix saved into error_dyn.

       If you believe this is not a problem, you can try to increase max_imag_freq_counter
       from the 'SSCHA_Minimizer' object or reduce the 'min_step_dyn'
Kindly help me solving this.
Here I am attaching my input file.
[sschapdh.zip](https://github.com/user-attachments/files/16302377/sschapdh.zip)

**Comment by mesonepigreco** [comment, marked as answer]
Hi,
It seems you are using the EMT calculator from ASE for a different material (PdH) than gold. While it is possible in principle to use it, I do not think results will be meaningful: it is possible that the forces and energies (as well as the harmonic dynamical matrix) computed by the EMT calculator for PdH are very bad, and this may lead to errors during the minimization like the one you are observing (it means that the minimization brings the system to a nonphysical solution, and the automatic procedure is not able to solve the problem).
The solution to the error is usually to reduce the minimization step
```
minim.set_minimization_step(0.0001) # from 0.01 -> 0.0001
```
however, in your case it may arise from the use of a bad potential for PdH.

Does the same code works if you use Gold instead of PdH? 

If you want to study PdH, I recommend either finding a potential specifically parameterized for PdH, or using ab-initio simulations (tutorial 1 and 2, but too much computationally expensive to run on a laptop).

**Comment by ArchSony** [comment]
Hi Sir,
Yes, I have tried and it works for the Gold system. In tutorial 7, the
tutor said that he did the sscha calculations and got the sscha_T0.0_dyn1
matrix. Can you please share the way he had done that?


*Thanks and Regards*
*Archana Hota*
*Junior Research Fellow*
*School of Mineral, Metallurgical and Materials Engineering*
*Indian Institute of Technology, Bhubaneswar*
*Odisha*



On Fri, Jul 19, 2024 at 12:15 PM Lorenzo Monacelli ***@***.***>
wrote:

> Hi,
> It seems you are using the EMT calculator from ASE for a different
> material (PdH) than gold. While it is possible in principle to use it, I do
> not think results will be meaningful: it is possible that the forces and
> energies (as well as the harmonic dynamical matrix) computed by the EMT
> calculator for PdH are very bad, and this may lead to errors during the
> minimization like the one you are observing (it means that the minimization
> brings the system to a nonphysical solution, and the automatic procedure is
> not able to solve the problem).
> The solution to the error is usually to reduce the minimization step
>
> minim.set_minimization_step(0.0001) # from 0.01 -> 0.0001
>
> however, in your case it may arise from the use of a bad potential for PdH.
>
> Does the same code works if you use Gold instead of PdH?
>
> If you want to study PdH, I recommend either finding a potential
> specifically parameterized for PdH, or using ab-initio simulations
> (tutorial 1 and 2, but too much computationally expensive to run on a
> laptop).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/SSCHAcode/discussions/347#discussioncomment-10092301>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJKNCU7Y55XDUVU2ZCUSJNDZNCYZFAVCNFSM6AAAAABLD4OCSSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMBZGIZTAMI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
*Disclaimer: *This email and any files transmitted with it are confidential 
and intended solely for the use of the individual or entity to whom they 
are addressed. If you have received this email in error please notify the 
system manager. This message contains confidential information and is 
intended only for the individual named. If you are not the named addressee 
you should not disseminate, distribute or copy this e-mail. Please notify 
the sender immediately by e-mail if you have received this e-mail by 
mistake and delete this e-mail from your system. If you are not the 
intended recipient you are notified that disclosing, copying, distributing 
or taking any action in reliance on the contents of this information is 
strictly prohibited.

**Reply by mesonepigreco** [reply]
The calculations on PdH were performed with ab initio DFT on a supercomputer (as explained in tutorial 1 and 2), the final dynamical matrix is the same as the one reported in the work [PRL, 111, 177002, 2013](https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.111.177002).

For this reason, we provide the final results of the SSCHA calculation in the tutorial. Since PdH is very cheap and symmetric, if you have a good workstation, you can probably still do it without the need for a supercomputer.

**Comment by ArchSony** [comment]
Thank you sir for the information!
*Thanks and Regards*
*Archana Hota*
*Junior Research Fellow*
*School of Mineral, Metallurgical and Materials Engineering*
*Indian Institute of Technology, Bhubaneswar*
*Odisha*



On Fri, Jul 19, 2024 at 1:23 PM Lorenzo Monacelli ***@***.***>
wrote:

> The calculations on PdH were performed with ab initio DFT on a
> supercomputer (as explained in tutorial 1 and 2), the final dynamical
> matrix is the same as the one reported in the work PRL, 111, 177002, 2013
> <https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.111.177002>.
>
> For this reason, we provide the final results of the SSCHA calculation in
> the tutorial. Since PdH is very cheap and symmetric, if you have a good
> workstation, you can probably still do it without the need for a
> supercomputer.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/SSCHAcode/discussions/347#discussioncomment-10092867>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJKNCU4IND2PEOOS4XIQK3LZNDAYXAVCNFSM6AAAAABLD4OCSSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMBZGI4DMNY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
*Disclaimer: *This email and any files transmitted with it are confidential 
and intended solely for the use of the individual or entity to whom they 
are addressed. If you have received this email in error please notify the 
system manager. This message contains confidential information and is 
intended only for the individual named. If you are not the named addressee 
you should not disseminate, distribute or copy this e-mail. Please notify 
the sender immediately by e-mail if you have received this e-mail by 
mistake and delete this e-mail from your system. If you are not the 
intended recipient you are notified that disclosing, copying, distributing 
or taking any action in reliance on the contents of this information is 
strictly prohibited.
